core/src/main/java/org/wildfly/glow/error/DatasourceErrorIdentification.java

```diff
@@ -130,32 +133,46 @@ public Map<Layer, Set<Env>> refreshErrors(Set<Layer> allBaseLayers) throws Excep
         if (unboundDatasourcesErrors != null) {
             for (IdentifiedError error : unboundDatasourcesErrors) {
                 UnboundDatasourceError uds = (UnboundDatasourceError) error;
-                for (Layer l : allBaseLayers) {
-                    if (l.getBringDatasources().contains(uds.unboundDatasource)) {
-                        // The error is directly handled, we can remove it.
-                        toRemove.add(uds.unboundDatasource);
-                        break;
-                    } else {
-                        if (l.getAddOn() != null) {
-                            Fix fix = l.getAddOn().getFixes().get(error.getId());
-                            if (fix != null) {
-                                String content = null;
-                                if (!l.getBringDatasources().contains(uds.unboundDatasource)) {
-                                    content = fix.getContent();
-                                    if (content != null) {
-                                        content = content.replaceAll("##ITEM##", uds.unboundDatasource);
-                                    }
-                                    if (fix.isEnv()) {
-                                        Set<Env> envs = ret.get(l);
-                                        if (envs == null) {
-                                            envs = new HashSet<>();
-                                            ret.put(l, envs);
+                if (datasourceDefinitionInfos.containsKey(uds.unboundDatasource)) {
+                    toRemove.add(uds.unboundDatasource);
+                    DataSourceDefinitionInfo di = datasourceDefinitionInfos.get(uds.unboundDatasource);
+                    if (di.getDriverLayer() == null) {
+                        Set<IdentifiedError> errs = errors.get(UNKNOWN_DRIVER_ERROR);
+                        if (errs == null) {
+                            errs = new HashSet<>();
+                            errors.put(UNKNOWN_DRIVER_ERROR, errs);
+                        }
```

Comment:
computeIfAbsent to reduce the complexity for a reader of the code

Reply:
Thank-you, fixed.